client/my-sites/github-deployments/components/repositories/create-repository/index.tsx

```diff
@@ -30,40 +28,69 @@ export const CreateRepository = () => {
 		page( indexPage( siteSlug! ) );
 	};
 
+	const canUserCreateDeployment = useSelector( ( state ) =>
+		canCurrentUser( state, siteId, 'manage_options' )
+	);
+
 	const dispatch = useDispatch();
 
-	const { createDeploymentAndRepository, isPending } = useCreateCodeDeploymentAndRepository(
-		siteId as number,
-		{
-			onSuccess: () => {
-				goToDeployments();
-				dispatch( successNotice( __( 'Deployment created.' ), noticeOptions ) );
-			},
-			onError: ( error ) => {
-				dispatch(
-					errorNotice(
-						// translators: "reason" is why connecting the branch failed.
-						sprintf( __( 'Failed to create repository: %(reason)s' ), { reason: error.message } ),
-						{
-							...noticeOptions,
-						}
-					)
-				);
-			},
-			onSettled: ( _, error ) => {
-				dispatch(
-					recordTracksEvent( 'calypso_hosting_github_create_repository_success', {
-						connected: ! error,
-					} )
-				);
-			},
-		}
-	);
+	const { createRepository, isPending } = useCreateRepository( {
+		onError: ( error ) => {
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why creating the repository failed.
+					sprintf( __( 'Failed to create repository: %(reason)s' ), { reason: error.message } ),
+					{
+						...noticeOptions,
+					}
+				)
+			);
+		},
+	} );
+	const { createDeployment, isPending: isDeploying } = useCreateCodeDeployment( siteId as number, {
+		onSuccess: () => {
+			goToDeployments();
+			dispatch( successNotice( __( 'Repository connected' ), noticeOptions ) );
+		},
+		onError: ( error ) => {
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why creating the repository failed.
+					sprintf( __( 'Failed to connect repository: %(reason)s' ), {
+						reason: error.message,
+					} ),
+					{
+						...noticeOptions,
+					}
+				)
+			);
+		},
+	} );
 
-	function handleCreateRepository( args: MutationVariables ) {
-		createDeploymentAndRepository( args );
+	if ( ! canUserCreateDeployment ) {
+		dispatch(
+			warningNotice( __( 'You do not have permissions to create a repository to this site.' ), {
+				isPersistent: true,
+			} )
+		);
 	}
```

Comment:
Ah that's true, I want to show the warning immediately as they load the page. useEffect it is

Reply:
From our p1708524646480559-slack-C06D9M3CHMK, The user will not be allowed to create repo if it cannot be connected, the purpose of creating a repo here is for deployments. Let's not allow creating a dangling repo if it's not gonna be connected, they can do that in github.

Reply:
Ok cool, let's redirect with a warning then. We don't need the check inside the mutation, too, in this case.

Reply:
@vykes-mac Another thing we can do is disable the button to create a new repo and add a popover saying that they're not allowed to do that 👌

Reply:
redirect feels a bit choppy to me rather than just showing the warning similar to when we try to modify domain contact but not eligible to do so. Domain contact info basically does what I'm doing at the moment.

Reply:
maybe it should be an error message instead 🔴

Reply:
@vykes-mac they won't be able to get there in the first place except for direct access; I don't think the redirect in case of direct access is too bad.